www/modules/custom/borg_mailers/borg_mailers_mailchimp/borg_mailers_mailchimp.module

```diff
@@ -1,11 +1,77 @@
 <?php
 /**
  * @file
- *   borg_mailers_mailchimp.module
- *   Conditionally Subscribe/Unsubscibe users to BackdropCMS.org mailing list on
- *   saving their user profile.
+ * Conditionally Subscribe/Unsubscibe to BackdropCMS.org mailing list on
+ * saving their account.
  */
 
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function borg_mailers_mailchimp_form_user_profile_form_alter(&$form, &$form_state) {
+  $uid = $form['#user']->uid;
+  $is_subcribed = db_query("SELECT mailchimp FROM {users} WHERE uid = :uid", array(':uid' => $uid))->fetchField();
+  $default = $is_subcribed ? TRUE : FALSE;
+  // Add a checkbox on the profile.
+  $form['mailchimp_signup'] = array(
```

Comment:
when you do this in this form alter does it automatically create the database tables?

Reply:
No those were added in the .install file (already tested and committed) it's the MailChimp part I needed reviewed. good point though, it would have made more sense for that to be in this PR too.